### PR TITLE
Update DirectXShaderCompiler to version 1.7.2308-o3de-rev2

### PR DESCRIPTION
**Pull request by akioCL**
-Add support for specialization constants

### CODE REVIEW

**Review by spham-amzn — APPROVED** [review]

**Comment by amzn-changml** [comment]
O3DE PR here: https://github.com/o3de/o3de/pull/18080. We'll run AR on this and promote to production after this passes, then merge this PR, and finally the O3DE PR

**Comment by amzn-changml** [comment]
This passed AR: https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-18080/14/pipeline. @akioCL Feel free to merge this PR when ready, and I'll promote it to prod

**Pull request merged**
